src/inputs/ObjectSearch.tsx

```diff
@@ -221,8 +228,15 @@ class ObjectSearch extends Component<IObjectSearchProps> {
     }
 
     // Select from search
-    const foundOption = this.options.find(option => option.id === selectedOption.key);
-    onChange(toJS(foundOption));
+    if (this.isMultiSelect) {
+      const foundOptions = this.options.filter(option => (
+        selectedOption.map((_selectedOption: any) => _selectedOption.key).includes(option.id)
```

Comment:
Might be nice to move this map outside of the filter loop, but this function isn't called on render so it's not a huge deal.